### PR TITLE
[NETBEANS-5048] PHP - cancel Find Usages action

### PR DESCRIPTION
**Pull request by KacerCZ**
https://issues.apache.org/jira/browse/NETBEANS-5048

Allows to cancel Find Usages action in PHP editor.

### CODE REVIEW

File: php/php.refactoring/src/org/netbeans/modules/refactoring/php/findusages/PhpWhereUsedQueryPlugin.java [hunk 1]
```diff
@@ -63,6 +64,10 @@ public Problem prepare(final RefactoringElementsBag elementsBag) {
                 if (fileObject == null) {
                     continue;
                 }
+                if (cancelled) {
+                    cancelled = false;
```
Comment (junichi11) [inline php/php.refactoring/src/org/netbeans/modules/refactoring/php/findusages/PhpWhereUsedQueryPlugin.java:68]: Maybe, need not this. `cancelled` is used only here at the moment, but I think we should keep the cancelled state.

Comment (KacerCZ) [reply php/php.refactoring/src/org/netbeans/modules/refactoring/php/findusages/PhpWhereUsedQueryPlugin.java:68]: If state is left, then it is impossible to repeat action by clicking green arrows in Usages window.

Comment (junichi11) [reply php/php.refactoring/src/org/netbeans/modules/refactoring/php/findusages/PhpWhereUsedQueryPlugin.java:68]: Ah, I see. You are right. Thanks.

File: php/php.refactoring/src/org/netbeans/modules/refactoring/php/findusages/PhpWhereUsedQueryPlugin.java [hunk 2]
```diff
@@ -63,6 +64,10 @@ public Problem prepare(final RefactoringElementsBag elementsBag) {
                 if (fileObject == null) {
                     continue;
                 }
+                if (cancelled) {
```
Comment (junichi11) [inline php/php.refactoring/src/org/netbeans/modules/refactoring/php/findusages/PhpWhereUsedQueryPlugin.java:67]: @tmysik Instead, should use `AtomicBoolean.compareAndSet()`?

Comment (junichi11) [reply php/php.refactoring/src/org/netbeans/modules/refactoring/php/findusages/PhpWhereUsedQueryPlugin.java:67]: We should leave a comment about why we change it to `false`.

Comment (KacerCZ) [reply php/php.refactoring/src/org/netbeans/modules/refactoring/php/findusages/PhpWhereUsedQueryPlugin.java:67]: Added comment why `cancelled` is reset.

For `cancelled` I used same type as in OccurencesFinderImpl in PHP Editor:
https://github.com/apache/netbeans/blob/514e170f43416aec1a25c1c443837bf1ef9043f0/php/php.editor/src/org/netbeans/modules/php/editor/csl/OccurrencesFinderImpl.java#L62

Comment (tmysik) [reply php/php.refactoring/src/org/netbeans/modules/refactoring/php/findusages/PhpWhereUsedQueryPlugin.java:67]: Volatile is OK here, IMHO.

**Review by tmysik — APPROVED** [review]

**Review by junichi11 — APPROVED** [review]

**Comment by KacerCZ** [comment]
@junichi11 @tmysik Thanks for review.

**Comment by tmysik** [comment]
@KacerCZ 

You are welcome. Thanks for your work!

**Pull request merged**